components/cam/src/dynamics/se/dyn_grid.F90

```diff
@@ -1174,7 +1538,116 @@ subroutine dyn_grid_get_elem_coords( ie, rlon, rlat, cdex )
     deallocate( clat, clon )
 
   end subroutine dyn_grid_get_elem_coords
-
-!#######################################################################
-
+  !
+  !=================================================================================================
+  !
+  subroutine fv_physgrid_init()
+    use control_mod,            only: cubed_sphere_map
+    use kinds,                  only: lng_dbl => longdouble_kind
```

Comment:
Why not use r8 precision here like everywhere else in the code? Unrelated, but I wasn't able to locate kinds_mod to see how longdouble is defined there, and definition does not seem to be part of this PR.

Reply:
I modeled this after things I noticed in the HOMME code. The kinds module is in:
    components/homme/src/share/kinds.F90

And it only really matters if HOMME_QUAD_PREC is defined, which seems to matter for calculations that convert between the reference coordinate and spherical coordinates. So that's why I used long_double_kind instead of r8.

<img src="https://user-images.githubusercontent.com/9679059/58964140-a5894d00-876b-11e9-81df-027cc0a6fe43.png"  width="400">

Reply:
Thanks! I did not realize cam is using kinds mod from homme folder (which is using some shr_kind_mod). It seems that your code is only active with CAM and homme_quad_prec is active only for standalone homme? If so, then long double here is the same as r8? Meanwhile I found shr_kind_mod, 4 of them, in cam/tools.

Reply:
I figure the quad precision could be used with CAM, but I'm not sure if that would ever be used. For current way we use the model it will be equivalent to r8.

Reply:
From grepping in cam folder, I see that a lot of code (if not all) uses r8 and none is using long double (please, correct me if this is wrong). Thus would it be better to switch to r8? Unless there is some future design in works that is going to use higher precision for grid calculations.

Reply:
My only concern is that these variables are consistent with the call to ref2sphere(). Looking back through this part it seems that I could use "real_kind" from the HOMME kinds module. I forget if this was also related to making the coordinates consistent with TempestRemap. But either way this needs to be consistent with the routines in HOMME rather than the rest of the CAM code. So if we change it we should use "real_kind" rather than "r8", even though they are effectively equivalent.